### ISSUE TITLE
Emitting no data for events

### ISSUE THREAD

**Issue by relu91** [w3c/wot-scripting-api#402]
Currently, emit event signature force to send a kind of InteractionInput which might be a DataSchema. Therefore, in my understanding, if I don't have any data for the event and since InteracitonInput is *not* optional, I should send `null` and event payload. Two questions:
- In JS it might be more idiomatic to have InteractionInput as optional -> the runtime will understand that it should not send any value
- "null" is still a valid JSON value so in principle, there might be corner cases where the developer would actually want to send "null" as a network value. How do we handle such use cases?

**Comment by JKRhb** [comment]
Leaving out the optional parameter would produce an `undefined` value, so maybe the developer could either do that or explicitly pass `undefined`? Then the two cases (`null` vs. `undefined`) could be distinguished.

**Comment by relu91** [comment]
The problem is that `undefined` is not a valid  `InteractionInput` value. Moreover, in WebIDL optional means most of the time that the value can be  `undefined`. So my proposal is to update the  `eventEmit` signature to:
```webidl
Promise<undefined> emitEvent(DOMString name, optional InteractionInput data )
```
which in TS translates to
```ts
async emitEvent(name: string, data?: InteractionInput)
```

**Comment by zolkis** [comment]
Sounds good, make a PR.
Technically [InteractionInput](https://w3c.github.io/wot-scripting-api/#the-interactioninput-type) is either a `ReadableStream` or any... which can be `undefined`.
However, it would be good to explicitly express optionality in the API.

**Comment by relu91** [comment]
> Technically [InteractionInput](https://w3c.github.io/wot-scripting-api/#the-interactioninput-type) is either a ReadableStream or any... which can be undefined.

Good catch I was using TS and in our type definitions DataSchemaValue (which in WebIDL is any) is more restrictive. See https://github.com/w3c/wot-scripting-api/blob/main/typescript/scripting-api/index.d.ts#L87 . I'll make a PR but now I wondering what we should do about that DataSchemaValue definition... I think we can discuss this in another Issue.

**Comment by danielpeintner** [comment]
Call 2022-05-30
* add `optional` AND check steps algorithm

**Comment by danielpeintner** [comment]
Call 2022-07-04
* we miss PR that changes
https://github.com/w3c/wot-scripting-api/blob/9294bc8899a2c578f3c7f12b32a65dfd1b9be500/typescript/scripting-api/index.d.ts#L291
to
` emitEvent(name: string, data?: InteractionInput): void;`
* check algorithm
